### PR TITLE
FOUNDATIONS: Throw Meaningful Exception

### LINKED ISSUE #2: FOUNDATIONS: SqlServerEFxceptionService Throw Meaningful Exception

### PR DESCRIPTION
**Pull request by JohnnySenior**
Closes #2 
@cjdutoit

### CODE REVIEW

**Review by ZafarUrakov — DISMISSED** [review]

File: STX.EFxceptions.SqlServer.Base/Models/Exceptions/DuplicateKeySqlException.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright(c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System;
+
+namespace STX.EFxceptions.SqlServer.Base.Models.Exceptions
+{
+    public class DuplicateKeySqlException : Exception
```
Comment (cjdutoit) [inline STX.EFxceptions.SqlServer.Base/Models/Exceptions/DuplicateKeySqlException.cs]:
```suggestion
    public class DuplicateKeySqlException : DbUpdateException
```

File: STX.EFxceptions.SqlServer.Base/Models/Exceptions/DuplicateKeyWithUniqueIndexSqlException.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright(c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System;
+
+namespace STX.EFxceptions.SqlServer.Base.Models.Exceptions
+{
+    public class DuplicateKeyWithUniqueIndexSqlException : Exception
```
Comment (cjdutoit) [inline STX.EFxceptions.SqlServer.Base/Models/Exceptions/DuplicateKeyWithUniqueIndexSqlException.cs]:
```suggestion
    public class DuplicateKeyWithUniqueIndexSqlException : DbUpdateException
```

File: STX.EFxceptions.SqlServer.Base/Models/Exceptions/ForeignKeyConstraintConflictSqlException.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright(c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System;
+
+namespace STX.EFxceptions.SqlServer.Base.Models.Exceptions
+{
+    public class ForeignKeyConstraintConflictSqlException : Exception
```
Comment (cjdutoit) [inline STX.EFxceptions.SqlServer.Base/Models/Exceptions/ForeignKeyConstraintConflictSqlException.cs]:
```suggestion
    public class ForeignKeyConstraintConflictSqlException : DbUpdateException
```

File: STX.EFxceptions.SqlServer.Base/Models/Exceptions/InvalidColumnNameSqlException.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright(c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System;
+
+namespace STX.EFxceptions.SqlServer.Base.Models.Exceptions
+{
+    public class InvalidColumnNameSqlException : Exception
```
Comment (cjdutoit) [inline STX.EFxceptions.SqlServer.Base/Models/Exceptions/InvalidColumnNameSqlException.cs]:
```suggestion
    public class InvalidColumnNameSqlException : DbUpdateException
```

File: STX.EFxceptions.SqlServer.Base/Models/Exceptions/InvalidObjectNameSqlException.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright(c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System;
+
+namespace STX.EFxceptions.SqlServer.Base.Models.Exceptions
+{
+    public class InvalidObjectNameSqlException : Exception
```
Comment (cjdutoit) [inline STX.EFxceptions.SqlServer.Base/Models/Exceptions/InvalidObjectNameSqlException.cs]:
```suggestion
    public class InvalidObjectNameSqlException : DbUpdateException
```

File: STX.EFxceptions.SqlServer.Base/Services/Foundations/SqlServerEFxceptionService.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+{
+    public partial class SqlServerEFxceptionService : ISqlServerEFxceptionService
+    {
+        private IDbErrorBroker<SqlException> sqlServerErrorBroker;
+
+        public SqlServerEFxceptionService(IDbErrorBroker<SqlException> sqlServerErrorBroker) =>
+            this.sqlServerErrorBroker = sqlServerErrorBroker;
+
+        public void ThrowMeaningfulException(DbUpdateException dbUpdateException)
+        {
```
Comment (cjdutoit) [inline STX.EFxceptions.SqlServer.Base/Services/Foundations/SqlServerEFxceptionService.cs:21]: Add a TryCatch here to wrap the local exceptions to the supported generic exceptions.  You will also need exception test for it.

**Review by cjdutoit — APPROVED** [review]

**Pull request merged**